internal/injector/imports.go

```diff
@@ -7,13 +7,28 @@ package injector
 
 import (
 	"go/token"
+	"slices"
 	"strconv"
 
+	"github.com/DataDog/orchestrion/internal/injector/aspect"
+	"github.com/DataDog/orchestrion/internal/injector/aspect/may"
 	"github.com/DataDog/orchestrion/internal/log"
 	"github.com/dave/dst"
 	"github.com/dave/dst/dstutil"
 )
 
+// packageFilterAspects filters out aspects that imply imports not present in the import map.
+func (i *Injector) packageFilterAspects(aspects []*aspect.Aspect) []*aspect.Aspect {
+	ctx := &may.PackageContext{
+		ImportPath: i.ImportPath,
+		ImportMap:  i.ImportMap,
+		TestMain:   i.TestMain,
+	}
+	return slices.DeleteFunc(aspects, func(a *aspect.Aspect) bool {
+		return a.JoinPoint.PackageMayMatch(ctx) == may.CantMatch
+	})
```

Comment:
The implementation is modifying the `aspects` slice in-place, so the documentation should mention this.

Alternatively -- the copy that is done upstream could be done in there instead; which would remove this concern.